src/components/TabularView.vue

```diff
@@ -0,0 +1,353 @@
+import queue from 'promise-the-world/queue.js'
+import { computed, defineEmits, defineProps, onMounted, ref, shallowRef, watch } from 'vue'
+import * as ns from '../namespace'
+import * as Remote from '../remote'
+import useLangStore, { observationsTermsWithNoLabel, shaclTermsWithNoLabel } from '../stores/langStore.js'
+import useViewStore from '../stores/viewStore.js'
+import { getBoundedViewPointer } from './common/debug.js'
+import { getEffectiveDimensions } from './common/labelFor.js'
+import { projectionFromView, updateViewProjection } from './common/projection.js'
+import DebugBox from './debug/DebugBox.vue'
+import DimensionHeader from './DimensionHeader.vue'
+import LoadingIcon from './icons/LoadingIcon.vue'
+import ObservationValue from './ObservationValue.vue'
+import PaginationMenu from './PaginationMenu.vue'
+
+const emit = defineEmits(['updateDataset'])
+
+const props = defineProps({
+  view: {
+    type: Object,
+    required: true
+  }
+})
+
+const page = ref()
+const pageSize = ref()
+const sortDimension = shallowRef()
+const sortDirection = shallowRef()
+
+const debugOpen = ref(false)
+const refreshCounter = ref(1)
+
+const queryQueue = queue(1)
+
+const viewStore = useViewStore()
+const {
+  currentView
+} = storeToRefs(viewStore)
+
+const langStore = useLangStore()
+const {
+  setPointers
+} = langStore
+
+const {
+  language,
+  pointer
+} = storeToRefs(langStore)
+
+// Used when projection changes
+const updateObservations = async () => {
+
+  if (currentView.value) {
+    let v = updateViewProjection({
+      view: currentView.value,
+      projection: {
+        page: page.value,
+        pageSize: pageSize.value,
+        sortDimension: sortDimension.value,
+        sortDirection: sortDirection.value,
+      }
+    })
+
+    await fetchObservationsAndLabels(v)
+  }
+}
+
+const observations = ref(Remote.loading())
+const observationCount = ref(Remote.loading())
+
+const fetchObservations = async (view) => {
+  observations.value = Remote.loading()
+  if (!view) return
+  await queryQueue.add(async () => {
+    try {
+      observations.value = await Remote.fetch(view.observations.bind(view))
+    } catch (error){
+      console.log(error)
+      console.log('view that produced the error')
+      console.log(getBoundedViewPointer(view).dataset.toString())
+    }
+  })
+  await queryQueue.add(async () => {
+    observationCount.value = await Remote.fetch(view.observationCount.bind(view))
+  })
+}
+
+function initProjection (view) {
+  // Get all the controls from the view
+  const projection = projectionFromView(view)
+  page.value = projection.page
+  pageSize.value = projection.pageSize
+  sortDimension.value = projection.sortDimension
+  sortDirection.value = projection.sortDirection
+}
+
+function refreshObservations () {
+  // Dirty hack since the view it's a shallow ref
+  // It's to update the observations component and show the labels when they arrive
+  refreshCounter.value = refreshCounter.value + 1
```

Comment:
There is probably a better way to trigger updates, but I haven't dig much into it.